src/screens/SearchScreen/components/DictionaryEntriesContainer.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import {ScrollView} from 'react-native';
+import PropTypes from 'prop-types';
+import DictionaryEntry from './DictionaryEntry.js';
+
+const DictionaryEntriesContainer = props => {
+  const updateInfo = entryNum => {
+    try {
+      return props.data[entryNum].targets[0].lemma;
+    } catch (error) {
+      return error.toString();
+    }
+  };
+
+  const getEntries = () => {
+    const entries = [];
+    for (entry in props.data)
+      entries.push(
+        <DictionaryEntry entryNum={entry} updateInfo={updateInfo} />,
+      );
+    return entries;
+  };
+
+  return <ScrollView>{getEntries()}</ScrollView>;
+};
+
+export default DictionaryEntriesContainer;
+
+DictionaryEntriesContainer.propTypes = {
+  data: PropTypes.array.isRequired,
```

Comment:
Might be worth expanding this into an `arrayOf` and defining some shape of the items contained.

Reply:
(Also, it looks like `data` can also be a string, when a bad search occurs?)

Reply:
To add to Doug's point - things get complicated if a prop can be more than 1 type of data although on occasion it is the best choice. Consider passing an `emptyMessage` as a separate prop and making `data` always be present and always be a (possibly empty) array.